openpype/lib/execute.py

```diff
@@ -198,7 +200,7 @@ def run_openpype_process(*args, **kwargs):
     args = get_openpype_execute_args(*args)
     env = kwargs.pop("env", None)
     # Keep env untouched if are passed and not empty
-    if not env:
+    if env is None:
```

Comment:
This should stay
```suggestion
    if not env:
```

Reply:
Why? :)

That way you could never pass an empty `env` which is exactly why passing `env = {}` looked weird in ExtractBurnin and color spaces file, it basically ended up still taking the local environment.

I've looked over the code base and those were the only places this function was being used.

I'd argue that passing `env = {}` should correctly run the subprocess with the empty environment.

Reply:
Main reason is backwards compatibility (we use the function in custom client addons). But even then, empty dictionary should be filled with os.environ and remove possible clashes.

Reply:
Why would an empty env be filled with `os.environ`. 

Why deviate from how subprocess works? Subprocess `env=None` copies `os.environ`. Passing `env={}` does exactly what you'd expect. It passes an empty environment.

Reply:
> It passes an empty environment.

The behavior is not the same in all Python versions (e.g. Maya 2020), sometimes is different per platform (I think it was 3.7.5?).

Reply:
I have yet to see a case where that happens, but even if it were it'd be a Python bug? The subprocess docs state that if it is not None it must be:

> If env is not None, it must be a mapping that defines the environment variables for the new process; these are used instead of the default behavior of inheriting the current process’ environment. It is passed directly to Popen. This mapping can be str to str on any platform or bytes to bytes on POSIX platforms much like os.environ or os.environb

Also there are many examples online showing subprocess with passing an empty dict as en to have an empty environment.